### ISSUE TITLE
Requests to `GET /api/card/123` is making slow queries on larger instances

### ISSUE THREAD

**Issue by flamber** [metabase/metabase#18759]
**Describe the bug**
Requests to `GET /api/card/123` are noticeably slower on 1.41.1 compared to 1.40.5
```
1.41.1: DEBUG middleware.log :: GET /api/card/123 200 1.9 s (10 DB calls)
1.40.5: DEBUG middleware.log :: GET /api/card/123 200 13.9 ms (7 DB calls)
```

There's 3 new statements in 1.41, when requesting `/api/card/123`:
```
SELECT avg("running_time") FROM "query_execution" WHERE ("running_time" IS NOT NULL AND "cache_hit" <> TRUE AND "card_id" = 123)
SELECT max("started_at") FROM "query_execution" WHERE ("running_time" IS NOT NULL AND "cache_hit" <> TRUE AND "card_id" = 123)
SELECT * FROM "moderation_review" WHERE ("moderated_item_type" = $1 AND ("moderated_item_id" in (123))) ORDER BY "id" DESC
  parameters: $1 = 'card'
```

But the first statement does not seem to use indexes, so that alone can take more than a second.
```
EXPLAIN ANALYZE SELECT avg("running_time") FROM "query_execution" WHERE ("running_time" IS NOT NULL AND "cache_hit" <> TRUE AND "card_id" = 123);
                                                                  QUERY PLAN
-----------------------------------------------------------------------------------------------------------------------------------------------
 Finalize Aggregate  (cost=157650.20..157650.21 rows=1 width=32) (actual time=919.797..921.154 rows=1 loops=1)
   ->  Gather  (cost=157649.98..157650.19 rows=2 width=32) (actual time=919.591..921.141 rows=3 loops=1)
         Workers Planned: 2
         Workers Launched: 2
         ->  Partial Aggregate  (cost=156649.98..156649.99 rows=1 width=32) (actual time=914.800..914.801 rows=1 loops=3)
               ->  Parallel Seq Scan on query_execution  (cost=0.00..156649.19 rows=317 width=4) (actual time=423.327..914.790 rows=1 loops=3)
                     Filter: ((running_time IS NOT NULL) AND (NOT cache_hit) AND (card_id = 123))
                     Rows Removed by Filter: 2243598
 Planning Time: 0.197 ms
 Execution Time: 921.206 ms

EXPLAIN ANALYZE SELECT max("started_at") FROM "query_execution" WHERE ("running_time" IS NOT NULL AND "cache_hit" <> TRUE AND "card_id" = 123);
                                                                                  QUERY PLAN
------------------------------------------------------------------------------------------------------------------------------------------------------------
 Result  (cost=362.37..362.38 rows=1 width=8) (actual time=8.262..8.263 rows=1 loops=1)
   InitPlan 1 (returns $0)
     ->  Limit  (cost=0.43..362.37 rows=1 width=8) (actual time=8.257..8.258 rows=1 loops=1)
           ->  Index Scan Backward using idx_query_execution_started_at on query_execution  (cost=0.43..275434.51 rows=761 width=8) (actual time=8.256..8.25
6 rows=1 loops=1)
                 Index Cond: (started_at IS NOT NULL)
                 Filter: ((running_time IS NOT NULL) AND (NOT cache_hit) AND (card_id = 123))
                 Rows Removed by Filter: 23518
 Planning Time: 0.238 ms
 Execution Time: 8.283 ms
```

**Expected behavior**
All controlled queries, which are executed often, should always have indexes.

**Information about your Metabase Installation:**
Tested 1.41.1 on Postgres 12 with a few thousands questions and executions - regression since 1.41.0

**Comment by camsaul** [comment]
There's definitely not an index on `card_id`

```
metabase=# \d query_execution;
                                        Table "public.query_execution"
    Column    |           Type           | Collation | Nullable |                   Default
--------------+--------------------------+-----------+----------+---------------------------------------------
 id           | integer                  |           | not null | nextval('query_execution_id_seq'::regclass)
 hash         | bytea                    |           | not null |
 started_at   | timestamp with time zone |           | not null |
 running_time | integer                  |           | not null |
 result_rows  | integer                  |           | not null |
 native       | boolean                  |           | not null |
 context      | character varying(32)    |           |          |
 error        | text                     |           |          |
 executor_id  | integer                  |           |          |
 card_id      | integer                  |           |          |
 dashboard_id | integer                  |           |          |
 pulse_id     | integer                  |           |          |
 database_id  | integer                  |           |          |
 cache_hit    | boolean                  |           |          |
Indexes:
    "query_execution_pkey" PRIMARY KEY, btree (id)
    "idx_query_execution_query_hash_started_at" btree (hash, started_at)
    "idx_query_execution_started_at" btree (started_at)

```

to fix this we should maybe just add the index on `card_id`

**Comment by camsaul** [comment]
We should probably also add an index on `moderation_review` `moderated_item_type` + `moderated_item_id` so finding specific moderation entries for a given object is fast

**Comment by camsaul** [comment]
This will be fixed by #18798 and #18799, but I think we will probably end up shipping these as part of 0.42.0 since we usually try to avoid introducing new database migrations in patch releases

**Comment by camsaul** [comment]
Fixed by #18798 and #18799
